kclvm/tools/src/LSP/src/document_symbol.rs

```diff
@@ -0,0 +1,96 @@
+use kclvm_ast::ast::Program;
+use kclvm_sema::resolver::scope::ProgramScope;
+use kclvm_sema::resolver::scope::Scope;
+use kclvm_sema::resolver::scope::ScopeKind;
+use kclvm_sema::resolver::scope::ScopeObject;
+use kclvm_sema::resolver::scope::ScopeObjectKind;
+use lsp_types::Range;
+use lsp_types::{DocumentSymbol, DocumentSymbolResponse, SymbolKind};
+
+use crate::to_lsp::lsp_pos;
+
+pub(crate) fn document_symbol(
+    file: &str,
+    _program: &Program,
+    prog_scope: &ProgramScope,
+) -> Option<lsp_types::DocumentSymbolResponse> {
+    let mut documentsymbols: Vec<DocumentSymbol> = vec![];
+    let scope = prog_scope.scope_map.get("__main__").unwrap().borrow();
+    // Get variable in scope
+    for obj in scope.elems.values().filter(|obj| {
+        obj.borrow().start.filename == file
```

Comment:
Using filename directly here is not a safe approach because of factors such as relative and absolute paths, as well as considering virtual files.

Reply:
The file location information of KCL Pos and LSP parameters are absolute paths. And, if there is a problem here, some symbols will be missing but will not cause errors.